specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1197 @@
+    };
+
+    runtimeclass AppNotificationButton
+    {
+        // Button can use ToolTip instead of content which requires empty content.
+        // Argument attribute is required and uses the ArgumentSerializer construct
+        // to add arguments to the button.
+        AppNotificationButton();
+        AppNotificationButton(String content);
+
+        AppNotificationButton AddArgument(String key, String value);
+
+        // Sets the IconUri for the button.
+        AppNotificationButton SetIconUri(Windows.Foundation.Uri iconUri);
+
+        // The tooltip for a button, if the button has an empty content string.
+        AppNotificationButton SetToolTip(String toolTip);
+
+        // Sets the Button as context menu action.
+        AppNotificationButton UseContextMenuPlacement();
+
+        // Sets the ButtonStyle to Success or Critical
+        AppNotificationButton SetButtonStyle(ButtonStyle buttonStyle);
+
+        // Sets the ID of an existing TextBox in order to have this button display to the right of the input.
+        AppNotificationButton SetInputId(String inputId);
+
+        // Launches the URI passed into the button when activated.
+        AppNotificationButton SetProtocolActivation(Windows.Foundation.Uri protocolUri);
+        AppNotificationButton SetProtocolActivation(Windows.Foundation.Uri protocolUri, String targetApplicationPfn);
+
+        // Retrieves the XML content of the button.
+        String GetXml();
+    };
+
+    runtimeclass AppNotificationComboBox
+    {
+        AppNotificationComboBox(String id);
+
+        // Add a selection item to the AppNotificationComboBox if InputType is AppNotificationComboBox. The id parameter is used to retrieve the user input when the app is activated
+        // and content is the displayed text for the item.
+        AppNotificationComboBox AddSelection(String id, String content);
+
+        // Adds a title to display on top
+        AppNotificationComboBox SetTitle(String title);
+
+        // Sets the default selection to be displayed by the AppNotificationComboBox
+        AppNotificationComboBox SetDefaultSelection(String id);
+
+        // Retrieves the XML content of the input.
+        String GetXml();
+    };
+
+    runtimeclass AppNotificationProgressBar
+    {
+        // AppNotificationProgressBar binds to AppNotificationProgressData so the AppNotification will
+        // receive every update to the status and value. In the WinAppSDK, these binding
+        // values are static, so developers won't need to define these binding values
+        // themselves.
+        AppNotificationProgressBar();
+
+        // Adds the title binding value.
+        AppNotificationProgressBar UseTitle();
+
+        // Adds the valueStringOverride value.
+        AppNotificationProgressBar UseValueStringOverride();
+
+        // Retrieves the XML content of the AppNotificationProgressBar
+        String GetXml();
+    };
+
+    enum MSWinSoundEvent
```

Comment:
`MS` is a suspicious prefix

Reply:
I tried matching the list on the schema: https://docs.microsoft.com/en-us/uwp/schemas/tiles/toastschema/element-audio

They all have ms-winsoundevent:Notification.* as the format. Would you prefer WinSoundEvent?

Reply:
Agree the naming here isn't good.

Reply:
Maybe call this AppNotificationSoundEvent?